Service/Scanner.php

```diff
@@ -7,45 +7,36 @@
 
 namespace Element119\SansecComposerIntegrityChecker\Service;
 
+use DI\Container;
 use Magento\Framework\App\Filesystem\DirectoryList;
 use Magento\Framework\Exception\FileSystemException;
-use Symfony\Component\Process\Process;
+use Sansec\Integrity\PackageResolver\LockReaderStrategy;
+use Sansec\Integrity\PackageSubmitter;
+use Symfony\Component\Console\Output\NullOutput;
 
 class Scanner
 {
-    public function __construct(private readonly DirectoryList $directoryList) {}
+    public function __construct(
+        private readonly Container $diContainer,
+        private readonly DirectoryList $directoryList,
+    ) {}
 
     /**
      * @return array
      * @throws FileSystemException
      */
     public function scan(): array
     {
-        $output = '';
-        $process = new Process($this->getCommandOptionsArray());
-        $process->run();
-
-        foreach ($process as $data) {
-            $output .= $data;
-        }
-
-        $output = substr($output, 0, strpos($output, ']') + 1); // json output only
-
-        return json_decode($output, true);
-    }
-
-    /**
-     * @throws FileSystemException
-     */
-    public function getCommandOptionsArray(): array
-    {
-        return [
-            'composer',
-            'integrity',
-            '--json',
-            '--no-ansi',
-            '--no-interaction',
-            sprintf('--working-dir=%s', $this->directoryList->getPath(DirectoryList::ROOT)),
-        ];
+        $scanner = $this->diContainer->make(
```

Comment:
FWIW I'm using PHP-DI container because I don't want to be reliant on Magento DI.

But since this is a Magento module, you probably just want to use the standard ways like injecting `PackageSubmitterFactory` etc.